run_cert.py

```diff
@@ -0,0 +1,333 @@
+        elif dns_provider == "route53":
+            if len(os.environ.get("AWS_ACCESS_KEY_ID", "")) == 0 or len(
+                    os.environ.get("AWS_SECRET_ACCESS_KEY", "")) == 0:
+                raise Exception("AWS_ACCESS_KEY_ID or AWS_SECRET_ACCESS_KEY is not defined")
+        else:
+            raise Exception("Unknown DNS verification method")
+
+
+def get_domains():
+    """Retrieve list of domains from own app definition or from environment variable based on verification method"""
+    data = get_marathon_app(os.environ.get(ENV_MARATHON_APP_ID))
+    verification_method = os.environ.get(ENV_VERIFICATION_METHOD, "http")
+    if verification_method == "http":
+        return data["app"]["labels"]["HAPROXY_0_VHOST"]
+    elif verification_method == "dns":
+        return os.environ.get(ENV_DOMAINS)
+    else:
+        raise Exception("Unknown verification method: " + verification_method)
+
+
+def get_cert_filepath(domain_name):
+    """Retrieve the certificate path based on the domain name"""
+    if domain_name.startswith("*"):
+        domain_name = domain_name.replace("*.", "")
+    latest_cert_dir = find_newest_dir(domain_name)
+    return f"{latest_cert_dir}/{domain_name}.pem"
+
+
+def update_marathon_app(app_id, **kwargs):
+    """Post new certificate data (as environment variable) to marathon to update the marathon-lb app definition"""
+    print("Uploading certificates", flush=True)
+    data = dict(id=app_id)
+    for key, value in kwargs.items():
+        data[key] = value
+    headers = {'Content-Type': 'application/json'}
+    response = requests.put(
+        "%(marathon_url)s/v2/apps/%(app_id)s" % dict(marathon_url=get_marathon_url(), app_id=app_id),
+        headers=headers, data=json.dumps(data), verify=False)
+    if not response.ok:
+        print(response)
+        print(response.text, flush=True)
+        raise Exception("Could not update app. See response text for error message.")
+    data = response.json()
+    # if not "deploymentId" in data:
+    if "deploymentId" not in data:
+        print(data, flush=True)
+        raise Exception("Could not update app. Marathon did not return deployment id. Please see error message.")
+    deployment_id = data['deploymentId']
+
+    # Wait for deployment to complete
+    deployment_exists = True
+    sum_wait_time = 0
+    while deployment_exists:
+        time.sleep(5)
+        sum_wait_time += 5
+        print("Waiting for deployment to complete", flush=True)
+        # Retrieve list of running deployments
+        response = requests.get("%(marathon_url)s/v2/deployments" % dict(marathon_url=get_marathon_url()), verify=False)
+        deployments = response.json()
+        deployment_exists = False
+        for deployment in deployments:
+            # Check if our deployment is still in the list
+            if deployment['id'] == deployment_id:
+                deployment_exists = True
+                break
+        if deployment_exists and sum_wait_time > 60 * 5:
```

Comment:
I would turn the 5 magic number into a "sleep_time" variable or something instead of repeating it 3 times on 206, 207, and 218.  I see it is not your code.